rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/interpreter/JarikoCallbackTest.kt

```diff
@@ -773,6 +773,134 @@ class JarikoCallbackTest : AbstractTest() {
         executeSourceLineTest("ERROR44")
     }
 
+    @Test
+    fun traceEmittedTest() {
+        val startTraces = mutableListOf<JarikoTrace>()
+        var finishCount = 0
+
+        val systemInterface = JavaSystemInterface().apply { onDisplay = { _, _ -> run {} } }
+        val configuration = Configuration().apply {
+            jarikoCallback.startJarikoTrace = { trace ->
+                startTraces.add(trace)
+            }
+            jarikoCallback.finishJarikoTrace = {
+                finishCount += 1
+            }
+        }
+        executePgm("TRACETST1", configuration = configuration, systemInterface = systemInterface)
+
+        assert(startTraces.isNotEmpty())
+        assert(finishCount > 0)
+    }
+
+    @Test
+    fun traceOpenedAlsoClosedTest() {
+        val startTraces = mutableListOf<JarikoTrace>()
+        var finishCount = 0
+
+        val systemInterface = JavaSystemInterface().apply { onDisplay = { _, _ -> run {} } }
+        val configuration = Configuration().apply {
+            jarikoCallback.startJarikoTrace = { trace ->
+                startTraces.add(trace)
+            }
+            jarikoCallback.finishJarikoTrace = {
+                finishCount += 1
+            }
+        }
+        executePgm("TRACETST1", configuration = configuration, systemInterface = systemInterface)
+        assertEquals(startTraces.size, finishCount)
+    }
+
+    @Test
+    fun traceSymtblTest() {
+        val traces = mutableListOf<JarikoTrace>()
+        val systemInterface = JavaSystemInterface().apply { onDisplay = { _, _ -> run {} } }
+        val configuration = Configuration().apply {
+            jarikoCallback.startJarikoTrace = { trace ->
+                traces.add(trace)
+            }
+        }
+        executePgm("TRACETST1", configuration = configuration, systemInterface = systemInterface)
+        val symtblTraces = traces.filter { it.kind == JarikoTraceKind.SymbolTable }
+
+        // (INIT + LOAD) * (n.called programs = 2)
+        assertEquals(symtblTraces.size, 4)
+
+        assert(symtblTraces.filter { it.description == "INIT" }.size == 2)
+        assert(symtblTraces.filter { it.description == "LOAD" }.size == 2)
+    }
+
+    @Test
+    fun traceCompositeStatementTest() {
+        val traces = mutableListOf<JarikoTrace>()
+        val systemInterface = JavaSystemInterface().apply { onDisplay = { _, _ -> run {} } }
+        val configuration = Configuration().apply {
+            jarikoCallback.startJarikoTrace = { trace ->
+                traces.add(trace)
+            }
+        }
+        executePgm("TRACETST1", configuration = configuration, systemInterface = systemInterface)
+        val compositeTraces = traces.filter { it.kind == JarikoTraceKind.CompositeStatement }
+
+        // 2 for each called program
+        assertEquals(compositeTraces.size, 4)
```

Comment:
Add an assertion on the description as well, as it is necessary for me to determine if the description makes sense.